### DISCUSSION TITLE
Error: No match for {"name":"JobPage","params":{"id":"1"}}

### DISCUSSION THREAD

**Discussion by Wolverin-e** [posva/unplugin-vue-router#434] [Questions and Help] [answered]
I am getting an Error while
-  <RouterLink :to="{ name: 'JobPage', params: { id: job.id }" }>Some Text<\/RouterLink>
- router.push({ name: 'JobPage', params: { id: job.id })

Error:
<img width="550" alt="image" src="https://github.com/posva/unplugin-vue-router/assets/43853684/5f4ddd24-d0b6-4810-8dee-a23ae40d3177">

Job ID Page has:
<img width="411" alt="image" src="https://github.com/posva/unplugin-vue-router/assets/43853684/77ff0b63-d0b8-4b2d-a966-918d241d8b06">

Pages Structure:
<img width="291" alt="image" src="https://github.com/posva/unplugin-vue-router/assets/43853684/2e42d5c5-4539-4f66-b82c-809b981119a3">

How I am using it:
<img width="825" alt="image" src="https://github.com/posva/unplugin-vue-router/assets/43853684/83128f61-246e-4c0c-83b2-9df8b7a049fa">

I am using
- "unplugin-vue-router": "^0.10.0",
- "vue-router": "^4.4.0"

Can someone please help me, how to resolve this?

**Comment by Wolverin-e** [comment]
Never Mind, It started working after I restarted the server.

**Reply by posva** [reply, marked as answer]
Oh I think I forgot to update the docs and this might be related to HMR. It's now better to add this:

```ts
import { createRouter, createWebHistory } from 'vue-router'
import { routes, handleHotUpdate } from 'vue-router/auto-routes'

export const router = createRouter({
  history: createWebHistory(),
  routes,
})

if (import.meta.hot) {
  handleHotUpdate(router)
}
```

And it should work without restarting or reloading the page

**Reply by posva** [reply]
I pushed an hmr page with this

**Reply by Wolverin-e** [reply]
Thanks Eduardo for the quick reply & this library,
I like using it 👍
